internal/process/snapshot.go

```diff
@@ -4,109 +4,205 @@
 package process
 
 import (
+	"math"
 	"strconv"
 
-	"github.com/newrelic/nri-vsphere/internal/config"
-
+	"github.com/newrelic/infra-integrations-sdk/data/metric"
 	"github.com/newrelic/infra-integrations-sdk/integration"
-	"github.com/vmware/govmomi/vim25/types"
 
-	"github.com/newrelic/infra-integrations-sdk/data/metric"
+	"github.com/sirupsen/logrus"
+	"github.com/vmware/govmomi/vim25/types"
 )
 
+const invalidFile = -1
+
+type snapshotProcessor struct {
```

Comment:
just if make sense, could you document really shortly how snapshots are orginized , or add some link ?